SMLHelper/Patchers/OptionsPanelPatcher.cs

```diff
@@ -94,5 +110,167 @@ internal static void AddTabs_Postfix(uGUI_OptionsPanel __instance)
                 }
             }
         }
+
+
+
+        // do not show tab if it is already visible (to prevent scroll position resetting)
+        internal static bool SetVisibleTab_Prefix(uGUI_TabbedControlsPanel __instance, int tabIndex)
+        {
+            return !(tabIndex >= 0 && tabIndex < __instance.tabs.Count && __instance.tabs[tabIndex].pane.activeSelf);
+        }
+
+        // adjusting ui elements
+        internal static void SetVisibleTab_Postfix(uGUI_TabbedControlsPanel __instance, int tabIndex)
+        {
+            if (tabIndex >= 0 && tabIndex < __instance.tabs.Count)
+            {
```

Comment:
CodeStyle + Readability: Invert if to reduce nesting.
```CS
if (tabIndex < 0 || tabIndex >= __instance.tabs.Count)
    return;

Transform options = __instance.tabs[tabIndex].container.transform;
....
```
Reference: https://en.wikipedia.org/wiki/De_Morgan's_laws